Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5680,14 +5680,33 @@ LayoutBoxExtent RenderBox::scrollPaddingForViewportRect(const LayoutRect& viewpo
 
 LayoutUnit synthesizedBaseline(const RenderBox& box, const RenderStyle& parentStyle, LineDirectionMode direction, BaselineSynthesisEdge edge)
 {
+    auto baselineType = [&] {
+        // https://drafts.csswg.org/css-inline-3/#alignment-baseline-property
+        // https://drafts.csswg.org/css-inline-3/#dominant-baseline-property
+
+        auto isHorizontalWritingMode = parentStyle.isHorizontalWritingMode();
+        auto textOrientation = parentStyle.textOrientation();
+
+        if (isHorizontalWritingMode || (!isHorizontalWritingMode && textOrientation == TextOrientation::Sideways))
```

Comment:
> if (isHorizontalWritingMode || (!isHorizontalWritingMode && textOrientation == TextOrientation::Sideways))

^^This looks slightly redundant.
`if (isHorizontalWritingMode || textOrientation == TextOrientation::Sideways)`